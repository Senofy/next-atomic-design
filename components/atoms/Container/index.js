import PropTypes from 'prop-types'
import styled from 'styled-components'
import { font, palette } from 'styled-theme'

export const Container = styled.div`
    position: relative;
    display: flex;
    flex-direction: column;
    min-width: 0;
    word-wrap: break-word;
    background-color: #fff;
    background-clip: border-box;
    border: 1px solid rgba(72, 94, 144, 0.16);
    border-radius: 0.25rem;
    padding: ${({ padding }) => `${padding}px`}
`

Container.propTypes = {
    reverse: PropTypes.bool,
}

Container.defaultProps = {
    palette: 'grayscale',
}