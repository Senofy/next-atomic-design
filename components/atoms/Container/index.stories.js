import { storiesOf } from '@storybook/react'
import { Container } from './'

storiesOf('Atom', module)
    .add('default', () => (
        <Container></Container>
    ))
