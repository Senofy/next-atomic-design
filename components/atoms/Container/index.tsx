import styled from 'styled-components';
import { color, padding } from 'styled-system';


export interface ContainerProps {
  padding: number;
}

export const Container = styled.div<ContainerProps>`
  position: relative;
  display: flex;
  flex-direction: column;
  min-width: 0;
  word-wrap: break-word;
  ${color};
  ${padding};
  background-clip: border-box;
  border: 1px solid rgba(72, 94, 144, 0.16);
  border-radius: 0.25rem;
`;