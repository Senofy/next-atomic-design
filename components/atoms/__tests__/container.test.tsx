import React from 'react'
import { render, cleanup } from '@testing-library/react'
import { Container } from '../Container'

afterEach(cleanup)

test('<Container> should take a snapshot', () => {
	const { asFragment } = render(<Container />)
	expect(asFragment()).toMatchSnapshot()
})

test('<Container> renders children when passed in', () => {
	const { getByText } = render(<Container>Hello</Container>)
	expect(getByText(/hello/i)).toBeInTheDocument()
})
