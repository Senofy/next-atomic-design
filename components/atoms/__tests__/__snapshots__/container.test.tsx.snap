// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Container> should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="Container-btg57n-0 iexaFt"
  />
</DocumentFragment>
`;
